core/src/main/scala/caliban/rendering/DocumentRenderer.scala

```diff
@@ -91,9 +114,13 @@ object DocumentRenderer extends Renderer[Document] {
     }
 
   private lazy val documentRenderer: Renderer[Document] = Renderer.combine(
-    directiveDefinitionsRenderer.contramap(_.directiveDefinitions),
+    (directiveDefinitionsRenderer ++
```

Comment:
Bit of a nitpick: I wonder if we should change `Renderer.Combined` to use a `Vector` instead, which has has `~O(1)` complexity over `List`'s `O(N)` for appending new elements. Or do we never have enough renderers combined to make a difference?

Reply:
I think that might be overkill, since for the most part renderers won't be constructed dynamically so the construction cost will be amortized.